packages/frameworks-sveltekit/src/lib/client.ts

```diff
@@ -63,7 +63,7 @@ export async function signIn<
   const data = await res.clone().json()
   const error = new URL(data.url).searchParams.get("error")
 
-  if (redirect || !isSupportingReturn || !error) {
+  if ((redirect && !error) || (!isSupportingReturn && !error)) {
```

Comment:
```suggestion
  if (redirect || !isSupportingReturn) {
```

Copied from https://github.com/nextauthjs/next-auth/blob/a0f3b04c4328533841105820bcc025eb00340132/packages/next-auth/src/react/index.tsx#LL269C1-L269C41

No need to check `!error` on both sides of an "||".